test/index.test.js

```diff
@@ -866,5 +866,27 @@ describe('mongoose module:', function() {
       const m = new mongoose.Mongoose();
       assert.deepEqual(await m.syncIndexes(), {});
     });
+    it('Allows a syncIndexes global option (gh-11030)', async function() {
+      const m = new mongoose.Mongoose();
+
+      const db = await m.connect('mongodb://localhost:27017/mongoose_test_11030');
+      const schema = new m.Schema({
+        title: { type: String, index: true }
+      });
+      const syncFalse = db.model('Sync', schema, 'Sync');
+      await syncFalse.collection.dropIndexes();
+      await syncFalse.init();
+      const Indexes = await syncFalse.listIndexes();
+      assert.equal(Indexes.length, 2);
+      await syncFalse.collection.createIndex({ name: 1 });
+      await m.deleteModel('Sync');
```

Comment:
I think this line seems to also force mongoose to drop the collection, which probably does not exist. thats why you get the ns not found error